From f5c120b68ae199288b35810f4cc4c6aca1b87ff6 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Thu, 31 Jul 2014 08:54:34 +0800
Subject: [PATCH] MA-6329 WFD between source and sink lost if disable USB
 tethering

In rndis_unbind(struct usb_configuration *c, struct usb_function *f),
it will free the rndis->notify_req and rndis.
But the rndis_response_available(void *_rndis) still will access
the rndis and still summit notify_req to our usb ep.
Add checking on used in rndis_msg_parser to avoid this case.

Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 drivers/usb/gadget/rndis.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/rndis.c b/drivers/usb/gadget/rndis.c
index d822d82..98df703 100644
--- a/drivers/usb/gadget/rndis.c
+++ b/drivers/usb/gadget/rndis.c
@@ -808,7 +808,8 @@ int rndis_msg_parser(u8 configNr, u8 *buf)
 	MsgType   = get_unaligned_le32(tmp++);
 	MsgLength = get_unaligned_le32(tmp++);
 
-	if (configNr >= RNDIS_MAX_CONFIGS)
+	if ((configNr >= RNDIS_MAX_CONFIGS) ||
+		(rndis_per_dev_params[configNr].used == 0))
 		return -ENOTSUPP;
 	params = &rndis_per_dev_params[configNr];
 
-- 
1.8.0

